### ISSUE TITLE
输入的句子最大长度是512吗？模型bge-large-zh-v1.5

### ISSUE THREAD

**Issue by Nightbringers** [FlagOpen/FlagEmbedding#199]
请问编码的句子长度最大是512吗？，模型bge-large-zh-v1.5。

**Comment by staoxiao** [comment]
是的，更长文本的模型在训练中。

**Comment by Nightbringers** [comment]
啥时候能出来，还有bge v2 不是说要出了吗，咋还没有，什么时候出

**Comment by Nightbringers** [comment]
encoded_input = tokenizer(sentences, padding=True, truncation=True, return_tensors='pt')
with torch.no_grad():
    model_output = model(**encoded_input)
    sentence_embeddings = model_output[0][:, 0]

还有这样使用是对的吧，用huggface， 编码出来的维度是1024

**Comment by staoxiao** [comment]
> 啥时候能出来，还有bge v2 不是说要出了吗，咋还没有，什么时候出

你这里的v2应该指的是[BAAI/llm-embedder](https://huggingface.co/BAAI/llm-embedder)，这是一个使用llama训练的英文检索器。
长文本的模型大概在12月发布。

**Comment by staoxiao** [comment]
> encoded_input = tokenizer(sentences, padding=True, truncation=True, return_tensors='pt') with torch.no_grad(): model_output = model(**encoded_input) sentence_embeddings = model_output[0][:, 0]
> 
> 还有这样使用是对的吧，用huggface， 编码出来的维度是1024

是的，使用方式可以参考https://github.com/FlagOpen/FlagEmbedding#usage-for-embedding-model

**Comment by Nightbringers** [comment]
请问有没有跟百度的rocketqa的效果做过对比呢

**Comment by staoxiao** [comment]
没有，rocketqa是一个网页检索模型，bge是通用向量模型，主要和mteb榜单上的模型对比。

**Comment by Nightbringers** [comment]
那如果我需要对一个问题检索有回答的段落是不是用rocketqa会好一些？

**Comment by staoxiao** [comment]
> 那如果我需要对一个问题检索有回答的段落是不是用rocketqa会好一些？

不同的模型在不同的领域下表现会有差异， 您可以实际对比后进行选择。

**Comment by mechigonft** [comment]
请问“请问编码的句子长度最大是512”，这个512指的是每个jsonl的字符长度吗？

**Comment by Nightbringers** [comment]
token的长度

**Comment by mechigonft** [comment]
哈喽，你好，详细请教下，请问是指的每条jsonl格式训练数据的token长度吗，还是指的jsonl里面的query的token长度，不包括pos和neg

**Comment by Nightbringers** [comment]
就是每条数据的token的长度， query，pos，neg 都是独立的，又不会把他们拼接起来，不管是 query，还是pos，还是neg ，编码时最大的token长度为512

**Comment by mechigonft** [comment]
好的，感谢

**Comment by mechigonft** [comment]
你好，我再深入请教一下，既然pos和neg都是数组形式，那这个token512是指的整个数组的token长度，还是数组里面每个元素的token长度

**Comment by Nightbringers** [comment]
都是文本形式，不是数组形式。每条文本的长度编码后最大长度不超过512 token

**Comment by mechigonft** [comment]
我摘取了作者的训练数据示例：
{"query": "Five women walk along a beach wearing flip-flops.", "pos": ["Some women with flip-flops on, are walking along the beach"], "neg": ["The 4 women are sitting on the beach.", "There was a reform in 1996.", "She's not going to court to clear her record.", "The man is talking about hawaii.", "A woman is standing outside.", "The battle was over. ", "A group of people plays volleyball."]}
其中neg是一个有7个字符串元素的数组，我想确认的是，token512指的是对单个字符串元素的长度限制，还是对整个数组内所有元素的长度和的限制？

**Comment by mechigonft** [comment]
抱歉，我可能有点啰嗦，因为我是这方面的新手，我想确认清楚，再进行训练

**Comment by Nightbringers** [comment]
你给的例子这里就是数组里的每个字符串token后的长度不超过512， 是对单个字符串的长度限制，不是整个数组

**Comment by staoxiao** [comment]
> 我摘取了作者的训练数据示例： {"query": "Five women walk along a beach wearing flip-flops.", "pos": ["Some women with flip-flops on, are walking along the beach"], "neg": ["The 4 women are sitting on the beach.", "There was a reform in 1996.", "She's not going to court to clear her record.", "The man is talking about hawaii.", "A woman is standing outside.", "The battle was over. ", "A group of people plays volleyball."]} 其中neg是一个有7个字符串元素的数组，我想确认的是，token512指的是对单个字符串元素的长度限制，还是对整个数组内所有元素的长度和的限制？

训练的时候以单个文本为单位，是对单个字符串元素的长度限制。对于长文本 ，代码会自动截断。

**Comment by mechigonft** [comment]
好的，十分感谢二位
